packages/headless/src/controllers/commerce/product-listing/headless-product-listing.ts

```diff
@@ -0,0 +1,132 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {CommerceAPIErrorStatusResponse} from '../../../api/commerce/commerce-api-error-response';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {configuration} from '../../../app/common-reducers';
+import {ProductRecommendation} from '../../../commerce.index';
+import {
+  fetchProductListing,
+  setProductListingUrl,
+} from '../../../features/product-listing/v2/product-listing-v2-actions';
+import {productListingV2Reducer as productListing} from '../../../features/product-listing/v2/product-listing-v2-slice';
+import {loadReducerError} from '../../../utils/errors';
+import {validateOptions} from '../../../utils/validate-payload';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+const optionsSchema = new Schema({
+  url: new StringValue({
+    required: true,
+    url: true,
```

Comment:
Is it an absolute requirement (from the API point of view) that this must be a valid URL that follows the regex defined in `string-value.ts` (in bueno) ?

If not, I would suggest chaning the check to `required: true` and `emptyAllowed: false`

Reply:
Good question, lemme check with the team :loading:

Reply:
We opted for only requiring the url to be set at all, as the format the API supports is quite lax!